src/pages/workspaces/workspace/useWorkspace.ts

```diff
@@ -62,17 +63,17 @@ export const useWorkspace = (namespace, name) : WorkspaceDetails => {
     workspaceStore.set(_.clone(workspace))
   }
 
-  const checkWorkspaceInitialization = workspace => {
+  const checkWorkspaceInitialization = (workspace, alreadyChecked = false) => {
```

Comment:
I think you can avoid introducing `alreadyChecked` by using the value of `workspaceInitialized`. In `WorkspaceContainer` we don't display the banner until `workspaceInitialized` is false; the variable goes from undefined -> false when we get back the first server call indicating that permissions are still syncing.

Reply:
The cleanest way to do this (that also avoids adding eventing to `useWorkspace`) probably is to add a `useEffect` hook in `WorkspaceContainer` that has `workspace.workspaceInitialized` as its dependency.